File: src/frontend/types/message.ts
export type MessageSender = "customer" | "ai" | "team" | "system";

export interface Message {
  id: string;
  conversationId: string;
  sender: MessageSender;
  senderName: string;
  content: string;
  timestamp: Date;
  read: boolean;
  attachments?: MessageAttachment[];
  metadata?: {
    aiGenerated?: boolean;
    needsApproval?: boolean;
    approved?: boolean;
    approvedBy?: string;
  };
}

export interface MessageAttachment {
  id: string;
  type: "image" | "document" | "video";
  url: string;
  filename: string;
  size: number;
}

export interface Conversation {
  id: string;
  customerId: string;
  customerName: string;
  customerPhone: string;
  messages: Message[];
  unreadCount: number;
  lastMessageAt: Date;
  status: "active" | "paused" | "archived";
  aiEnabled: boolean;
  assignedTo?: string;
  tags: string[];
  priority: "low" | "medium" | "high" | "urgent";
}

export interface InternalNote {
  id: string;
  conversationId: string;
  authorId: string;
  authorName: string;
  content: string;
  createdAt: Date;
  updatedAt?: Date;
}

File: src/backend/schemas/message.py
from uuid import UUID
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class MessageBase(BaseModel):
    sender: str  # customer, ai, team, system
    sender_name: str
    content: str


class MessageCreate(MessageBase):
    conversation_id: UUID
    ai_generated: bool = False
    needs_approval: bool = False
    twilio_sid: Optional[str] = None


class MessageRead(MessageBase):
    id: UUID
    conversation_id: UUID
    read: bool
    ai_generated: bool
    needs_approval: bool
    approved: Optional[bool] = None
    twilio_sid: Optional[str] = None
    created_at: datetime

    model_config = {"from_attributes": True}


class SendMessageRequest(BaseModel):
    content: str
    sender_name: str = "Jake Bennett"
